### PR TITLE
Create markdown-links-verifier.yml

### LINKED ISSUE #30886: Fix broken links
See GitHub Action run in https://github.com/dotnet/aspnetcore/pull/30885

*Let me know if there are any false positives.

Summary of the results:

```diff
+ Validating links in: /github/workspace/CONTRIBUTING.md.
- ##[error]Invalid link: '/docs/README.md' relative to '/github/workspace'.

+ Validating links in: /github/workspace/docs/BuildFromSource.md.
- ##[error]Invalid link: '/eng/scripts/InstallVisualStudio.ps1' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/global.json' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/scripts/vs.json' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/scripts/InstallJdk.ps1' relative to '/github/workspace/docs'.
- ##[error]Invalid link: 'using-dotnet-on-command-line-in-this-repo' relative to '/github/workspace/docs'.

+ Validating links in: /github/workspace/docs/SharedFramework.md.
- ##[error]Invalid link: '/eng/SharedFramework.Local.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/SharedFramework.External.props' relative to '/github/workspace/docs'.

+ Validating links in: /github/workspace/docs/ReferenceResolution.md.
- ##[error]Invalid link: '/eng/targets/ResolveReferences.targets' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/tools/BaselineGenerator/README.md' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Baseline.xml' relative to '/github/workspace/docs'.
- ##[error]Invalid link: 'eng/Baseline.Designer.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Dependencies.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/ProjectReferences.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Versions.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Version.Details.xml' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Versions.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: 'global.json' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Dependencies.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Versions.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/Version.Details.xml' relative to '/github/workspace/docs'.

+ Validating links in: /github/workspace/docs/PreparingPatchUpdates.md.
- ##[error]Invalid link: '/eng/Versions.props' relative to '/github/workspace/docs'.
- ##[error]Invalid link: '/eng/tools/BaselineGenerator/README.md' relative to '/github/workspace/docs'.

+ Validating links in: /github/workspace/src/Tools/README.md.
- ##[error]Invalid link: 'dotnet-watch/README.md' relative to '/github/workspace/src/Tools'.

+ Validating links in: /github/workspace/src/Html.Abstractions/README.md.
- ##[error]Invalid link: '../../../BuildFromSource.md#building-a-subset-of-the-code' relative to '/github/workspace/src/Html.Abstractions'.
- ##[error]Invalid link: '../../../BuildFromSource.md#running-tests-on-command-line' relative to '/github/workspace/src/Html.Abstractions'.
- ##[error]Invalid link: '../../../README.md' relative to '/github/workspace/src/Html.Abstractions'.

+ Validating links in: /github/workspace/src/Middleware/WebSockets/README.md.
- ##[error]Invalid link: './test/AutobahnTestApp/README.md' relative to '/github/workspace/src/Middleware/WebSockets'.


+ Validating links in: /github/workspace/src/Shared/ErrorPage/README.md.
- ##[error]Invalid link: '/src/Razor/tools/RazorSyntaxGenerator/' relative to '/github/workspace/src/Shared/ErrorPage'.

+ Validating links in: /github/workspace/src/SignalR/docs/GettingStarted.md.
- ##[error]Invalid link: '/docs/BuildFromSource.md' relative to '/github/workspace/src/SignalR/docs'.

+ Validating links in: /github/workspace/eng/tools/BaselineGenerator/README.md.
- ##[error]Invalid link: '/eng/Baseline.xml' relative to '/github/workspace/eng/tools/BaselineGenerator'.

+ Validating links in: /github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/README.md.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp'.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp'.

+ Validating links in: /github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp/README.md.
- ##[error]Invalid link: 'src/index.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: 'src/index.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: 'src/registerServiceWorker.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: 'src/registerServiceWorker.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
```

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #30886.

### CODE REVIEW

File: .github/workflows/markdown-links-verifier.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  validate_links:
+    name: Markdown links verifier
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout the repository
+      uses: actions/checkout@v1
+
+    - name: Validate links
+      uses: Youssef1313/markdown-links-verifier@v0.0.5
```
Comment (Youssef1313) [inline .github/workflows/markdown-links-verifier.yml:14]: I don't know your policy about using a 3rd-party GH Action. The results of broken links are there to fix anyway, whether you'll use the GH Action or not.

Comment (BrennanConroy) [reply .github/workflows/markdown-links-verifier.yml:14]: @blowdart

Comment (Youssef1313) [inline .github/workflows/markdown-links-verifier.yml:14]:
```suggestion
      uses: Youssef1313/markdown-links-verifier@v0.0.6
```

**Comment by Youssef1313** [comment]
Can someone look at the current invalid links?

```
+ Validating links in: /github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/README.md.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp'.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp'.

+ Validating links in: /github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp/README.md.
- ##[error]Invalid link: 'src/index.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: 'src/index.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: 'src/registerServiceWorker.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: 'src/registerServiceWorker.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.
- ##[error]Invalid link: '../config/webpack.config.prod.js' relative to '/github/workspace/src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp'.

```

**Comment by blowdart** [comment]
We just did this in corefx. 

@mmitche I believe our decision was we cannot have markdown lint running our build system because it installs too much stuff we simply can't audit, however inside a github action is fine. Is my memory correct on this?

**Comment by Youssef1313** [comment]
@blowdart This is not markdownlint (which is indeed used in dotnet/runtime).

**Comment by blowdart** [comment]
Oh I missed it's your own! 

We'd have to review it first, and it can't make it into the main builds, it'd also have to get pinned to each version we review. Onboarding new things have rules around it.

@BrennanConroy I'll let you discuss with everyone the benefits and then we can think about integrating this if you feel it adds a lot of value.

**Comment by Youssef1313** [comment]
Yeah I know it's not an easy decision to depend on it. Here is the [repo](https://github.com/Youssef1313/markdown-links-verifier/) in case you want to review. If it's not suitable for you to use this action, it's completely fine and I understand that. Just let me know and I'll remove the yml file and only fix the links.

**Comment by mmitche** [comment]
> We just did this in corefx.
> 
> @mmitche I believe our decision was we cannot have markdown lint running our build system because it installs too much stuff we simply can't audit, however inside a github action is fine. Is my memory correct on this?

Yeah I'm fine with this kind of thing running in an isolated action.

**Comment by javiercn** [comment]
@Pilchie should we discuss add this to the DoI agenda?

**Comment by Pilchie** [comment]
@javiercn That seems reasonable to me.

File: src/Tools/README.md [hunk 2]
```diff
@@ -4,7 +4,6 @@
 
 The folder contains command-line tools for ASP.NET Core. The following tools are bundled* in the .NET Core CLI. Follow the links below for more details on each tool.
 
-- [dotnet-watch](dotnet-watch/README.md)
```
Comment (Youssef1313) [inline src/Tools/README.md:7]: I didn't find dotnet-watch in this directory. So I deleted the link and updated usages of dotnet watch examples below.

**Comment by javiercn** [comment]
We've discussed this and we'll consider taking this change, however we need to go through some of the logistics.

**Comment by javiercn** [comment]
Unfortunately we've decided we don't plan to take this PR at this time.

**Pull request closed without merging**